### PR TITLE
Use ng2tc for compiling and running tests on ci and fix issues with it

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/core/src/reflection/reflection_capabilities.ts [hunk 1]
```diff
@@ -128,6 +128,16 @@ export class ReflectionCapabilities implements PlatformReflectionCapabilities {
     if (isPresent((<any>typeOrFunc).parameters)) {
       return (<any>typeOrFunc).parameters;
     }
+
+    // API of tsickle for lowering decorators to properties on the class.
+    if (isPresent((<any>typeOrFunc).ctorParameters)) {
+      let ctorParameters = (<any>typeOrFunc).ctorParameters;
+      let paramTypes = ctorParameters.map( ctorParam => ctorParam && ctorParam.type );
+      let paramAnnotations = ctorParameters.map( ctorParam => ctorParam && convertTsickleDecoratorIntoMetadata(ctorParam.decorators) );
```
Comment (alexeagle) [inline modules/@angular/core/src/reflection/reflection_capabilities.ts]: don't forget to format

File: tools/tsc-watch/tsc_watch.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ enum State {
 }
 
 export const TSC = 'node_modules/typescript/bin/tsc';
+export const NG2TSC = 'node_modules/.bin/ng2tc';
```
Comment (alexeagle) [inline tools/tsc-watch/tsc_watch.ts]: nit: NG2TSC doesn't match ng2tc

**Comment by alexeagle** [comment]
LG, needs rebase on my PR with the build.sh and tsconfig.json changes

File: tools/tsc-watch/tsc_watch.ts (hunk 2 above)
Comment (tbosch) [inline tools/tsc-watch/tsc_watch.ts]: updated (not pushed yet)

File: modules/@angular/core/src/reflection/reflection_capabilities.ts (hunk 1 above)
Comment (tbosch) [inline modules/@angular/core/src/reflection/reflection_capabilities.ts]: updated (not pushed yet)

**Pull request closed without merging**